### PR TITLE
Feature/Allow Hooks with in-memory db

### PR DESCRIPTION
**Pull request by kuznar**
Related discussion: https://github.com/sorry-cypress/sorry-cypress/discussions/719

## Purpose: 

Today if you wish to utilize the hooks (bitbucket, slack, web, etc) you must deploy and maintain the full project (dashboard, api, director, and a db). I would like to utilize the hook with just the director and it's in-memory db.

## Changes:

- Allow Hooks to initize when in-memory is used
- in-memory database tracks progress
- /hooks GET and POST endpoint to set hooks

### Setting hooks with API
Screenshot from 2023-01-15 12-56-35

### Output from cypress
Screenshot from 2023-01-15 13-19-13

### Output from webhooks (consumed by dummy express server)
```
port 3005 up
WEBHOOK RECEIVED: {
  event: 'RUN_START',
  runUrl: 'http://localhost:8080/run/a13bcbe25589d2021628ad6d33221b91',
  buildId: 'hello-cypress10',
  commit: {
    sha: 'f494fc2952259791210f28e0d4dc26bb94324479',
    branch: 'master',
    authorName: 'kuznar',
    authorEmail: '107443276+kuznar@users.noreply.github.com',
    message: 'dummy cypress tests\n',
    defaultBranch: null
  },
  overall: 0,
  passes: 0,
  failures: 0,
  skipped: 0,
  pending: 0,
  flaky: 0
}
WEBHOOK RECEIVED: {
  event: 'INSTANCE_FINISH',
  runUrl: 'http://localhost:8080/run/a13bcbe25589d2021628ad6d33221b91',
  buildId: 'hello-cypress10',
  commit: {
    sha: 'f494fc2952259791210f28e0d4dc26bb94324479',
    branch: 'master',
    authorName: 'kuznar',
    authorEmail: '107443276+kuznar@users.noreply.github.com',
    message: 'dummy cypress tests\n',
    defaultBranch: null
  },
  overall: 3,
  passes: 0,
  failures: 1,
  skipped: 2,
  pending: 0,
  flaky: 0,
  spec: 'cypress/e2e/1-getting-started/skipme.cy.js'
}
WEBHOOK RECEIVED: {
  event: 'INSTANCE_FINISH',
  runUrl: 'http://localhost:8080/run/a13bcbe25589d2021628ad6d33221b91',
  buildId: 'hello-cypress10',
  commit: {
    sha: 'f494fc2952259791210f28e0d4dc26bb94324479',
    branch: 'master',
    authorName: 'kuznar',
    authorEmail: '107443276+kuznar@users.noreply.github.com',
    message: 'dummy cypress tests\n',
    defaultBranch: null
  },
  overall: 6,
  passes: 1,
  failures: 3,
  skipped: 2,
  pending: 0,
  flaky: 0,
  spec: 'cypress/e2e/1-getting-started/todo.cy.js'
}
WEBHOOK RECEIVED: {
  event: 'INSTANCE_FINISH',
  runUrl: 'http://localhost:8080/run/a13bcbe25589d2021628ad6d33221b91',
  buildId: 'hello-cypress10',
  commit: {
    sha: 'f494fc2952259791210f28e0d4dc26bb94324479',
    branch: 'master',
    authorName: 'kuznar',
    authorEmail: '107443276+kuznar@users.noreply.github.com',
    message: 'dummy cypress tests\n',
    defaultBranch: null
  },
  overall: 10,
  passes: 5,
  failures: 3,
  skipped: 2,
  pending: 0,
  flaky: 0,
  spec: 'cypress/e2e/1-getting-started/todo2.cy.js'
}
WEBHOOK RECEIVED: {
  event: 'RUN_FINISH',
  runUrl: 'http://localhost:8080/run/a13bcbe25589d2021628ad6d33221b91',
  buildId: 'hello-cypress10',
  commit: {
    sha: 'f494fc2952259791210f28e0d4dc26bb94324479',
    branch: 'master',
    authorName: 'kuznar',
    authorEmail: '107443276+kuznar@users.noreply.github.com',
    message: 'dummy cypress tests\n',
    defaultBranch: null
  },
  overall: 10,
  passes: 5,
  failures: 3,
  skipped: 2,
  pending: 0,
  flaky: 0
}
```

### CODE REVIEW

**Review by agoldis — COMMENTED** [review]
Thank you!

What about all the `ts-ignore`s?

File: packages/director/src/execution/mongo/runCompletion/runCompletion.ts [hunk 1]
```diff
@@ -4,22 +4,14 @@ import { runTimeoutModel } from '@sorry-cypress/mongo';
...
   getNonCompletedGroups,
+  maybeSetRunCompleted,
 } from '../runs/run.controller';
 import {
   getRunById,
   setRunCompleted,
   setRunCompletedWithTimeout,
 } from '../runs/run.model';
 
-export const maybeSetRunCompleted = async (runId: string) => {
```
Comment (kuznar) [inline packages/director/src/execution/mongo/runCompletion/runCompletion.ts:14]: Looks like this was a dub.

**Comment by kuznar** [comment]
Hey Andrew, cleaned up those TS ignores and refactored a little. Changed it so the /hooks endpoint does not reply with all the hooks since it can have potentially sensitive tokens, and remove the GET endpoint. New response from POST is just status and message. 

Also raise PR for docs: https://github.com/sorry-cypress/gitbook/pull/22


Screenshot from 2023-01-18 18-46-11

**Comment by kuznar** [comment]
Unit tests are passing for me locally....

**Comment by agoldis** [comment]
@kuznar please sync with master

**Pull request merged**